src/python/pants/backend/java/target_types.py

```diff
@@ -84,8 +84,7 @@ async def generate_targets_from_junit_tests(
         request.generator,
         paths.files,
         union_membership,
-        # TODO(#12790): set to false when dependency inference is disabled.
-        add_dependencies_on_all_siblings=True,
+        add_dependencies_on_all_siblings=False,
```

Comment:
This should check what the subsystem is set to like this

https://github.com/pantsbuild/pants/blob/b8a406612420184dcd008a38f6eff35c516d1cad/src/python/pants/backend/python/target_types_rules.py#L87